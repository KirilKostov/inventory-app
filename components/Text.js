import React from "react";
import { Text as CustomText, StyleSheet } from "react-native";

export const Text = (props) => {
  return (
    <CustomText style={{ ...styles.text, ...props.style }}>
      {props.children}
    </CustomText>
  );
};

const styles = StyleSheet.create({
  text: {
    fontFamily: "gibson-regular",
    fontSize: 16,
  },
});
