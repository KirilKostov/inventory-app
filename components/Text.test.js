import React from "react";
import renderer from "react-test-renderer";

import { Text } from "./Text";

describe("<Text />", () => {
  it("renders correctly", () => {
    const testRenderer = renderer
      .create(<Text>Simple body text</Text>)
      .toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
