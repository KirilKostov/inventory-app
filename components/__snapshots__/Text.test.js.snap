// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<Text
  style={
    Object {
      "fontFamily": "gibson-regular",
      "fontSize": 16,
    }
  }
>
  Simple body text
</Text>
`;
